src/alire/alire-config.adb

```diff
@@ -0,0 +1,32 @@
+with Alire.Environment;
+with Alire.OS_Lib;
+with Alire.Platform;
+
+package body Alire.Config is
+
+   Config_Path : access String;
+   
+   ----------
+   -- Path --
+   ----------
+
+   function Path return String is
+   begin
+      if Config_Path /= null then -- Case with switch (TODO)
+         return Config_Path.all;
+      else
+         return OS_Lib.Getenv (Environment.Config, 
+                               Platform.Default_Config_Folder);
+      end if;
+   end Path;
+
+   procedure Set_Path (Path : String) is
+   begin
+      if Config_Path /= null then
+         raise Constraint_Error with "Custom path already set";
+      else
+         Config_Path := new String'(Path);
+      end if;
+   end Set_Path;
```

Comment:
In Alr.Platforms, a comment says that the config directory is used to store alr’s source code. Given that we’re moving away from recompilation, is the config directory still relevant?

Reply:
I also thought that we could get rid of it. Then I remembered a couple things we need the config folder for:
- Templates for initialization of projects (inside the source checkout in the other comment).
- The downloaded indexes (related: #69).

The source override is only meant for us developers, so we can tinker with resources under version control directly in the source tree (the aforementioned templates), if the need ever arises.

Reply:
Ah, good point! It would look more natural to me if the downloaded indexes went to the `cache` directory, but that’s for another time :)